homeassistant/components/group/__init__.py

```diff
@@ -87,35 +84,18 @@ def _conf_preprocess(value):
     extra=vol.ALLOW_EXTRA,
 )
 
-# List of ON/OFF state tuples for groupable states
-_GROUP_TYPES = [
-    (STATE_ON, STATE_OFF),
-    (STATE_HOME, STATE_NOT_HOME),
-    (STATE_OPEN, STATE_CLOSED),
-    (STATE_LOCKED, STATE_UNLOCKED),
-    (STATE_PROBLEM, STATE_OK),
-]
-
-
-def _get_group_on_off(state):
-    """Determine the group on/off states based on a state."""
-    for states in _GROUP_TYPES:
-        if state in states:
-            return states
-
-    return None, None
+ON_OFF_MAPPING = "on_off_mapping"
+ON_STATES_BY_DOMAIN = "on_states_by_domain"
+EXCLUDE_DOMAINS = "exclude_domains"
 
 
 @bind_hass
 def is_on(hass, entity_id):
     """Test if the group state is in its ON-state."""
     state = hass.states.get(entity_id)
 
-    if state:
-        group_on, _ = _get_group_on_off(state.state)
-
-        # If we found a group_type, compare to ON-state
-        return group_on is not None and state.state == group_on
+    if state is not None:
+        return state.state in hass.data[DATA_KEY][ON_OFF_MAPPING]
```

Comment:
This will require the group integration to be set up.

Reply:
Do we create groups in production without the group integration being setup?

Reply:
I guess it could be called before the groups are created?

Reply:
But then the state would be None?

Reply:
I added a test for this case.